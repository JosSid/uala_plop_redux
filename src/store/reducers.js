//  { ESQUEMA DE ESTADO
//      auth: true/false
//      ads: []
//      tags: []
//  }

import { ADS_LOADED, AUTH_LOGIN, AUTH_LOGOUT, TAGS_LOADED } from "./types";

const defaultState = {
    auth: false,
    ads: [],
    tags: []
};

export function auth(state = defaultState.auth, action) {
    if(action.type === AUTH_LOGIN){
        return true;
    };
    if(action.type === AUTH_LOGOUT){
        return false;
    };
    
    return state;
};

export function ads(state = defaultState.ads, action) {
    if(action.type === ADS_LOADED){
        return action.payload;
    };

    return state;
};

export function tags(state = defaultState.tags, action) {
    if(action.type === TAGS_LOADED){
        return action.payload
    };

    return state
};

// export default function reducer( state = defaultState, action) {
//     return {
//         auth: auth(state.auth, action),
//         ads: ads(state.ads, action)}
// };