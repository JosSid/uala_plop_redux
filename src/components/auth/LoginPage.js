import { useState, useEffect } from 'react';
import styles from './LoginPage.module.css';
import Button from '../common/Button.js';
import CheckBox from '../common/CheckBox.js';
import Form from '../common/form.js/Form';
import FormField from '../common/formField/FormField.js';
import storage from '../../utils/storage';
import ErrorDisplay from '../common/error/errorDisplay/ErrorDisplay.js';
import { useDispatch, useSelector } from 'react-redux';
import { authLogin, authLogout, uiResetError } from '../../store/actions';
import { getUi, getIsLogged } from '../../store/selectors';

const LoginPage = ({ titleApp }) => {
  const dispatch = useDispatch();
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [check, setCheck] = useState(false);
  const { isFetching, error } = useSelector(getUi);
  const isLogged = useSelector(getIsLogged)

  const handleChangeEMail = (event) => setEmail(event.target.value);
  const handleChangePassword = (event) => setPassword(event.target.value);

  const handleChangeChecked = (event) => setCheck(event.target.checked);

  const resetError = () => dispatch(uiResetError());

  const handleSubmit = async (event) => {
    event.preventDefault();
    const accessToken = await dispatch(authLogin({ email, password }));
    check && storage.set('auth', accessToken);
  };

  const isEnabledButton = () => email && password && !isFetching;

  useEffect(() => {
    isLogged && dispatch(authLogout());
  }, [dispatch, isLogged]);

  return (
    <div className={styles.loginPage}>
      <h1 className={styles.loginPage__title}>
        {'Welcome to'} <br />
        {`${titleApp}`}
      </h1>
      <h4 className={styles.loginPage__title}>
        Please,login for search or create your advertisment
      </h4>
   
      <Form initialValue={{email: '', password: ''}} onSubmit={handleSubmit} className={styles.loginPage__form} >
        <FormField
          type='text'
          name='username'
          label='eMail'
          className={styles.loginForm__field}
          onChange={handleChangeEMail}
          value={email}
        />

        <FormField
          type='password'
          name='password'
          label='password'
          className={styles.loginForm__field}
          onChange={handleChangePassword}
          value={password}
        />

        <Button
          type='submit'
          className={styles.loginForm__submit}
          disabled={!isEnabledButton()}
        >
          Log in
        </Button>
        <CheckBox
          name='checklog'
          type='checkbox'
          label='Check for recording Login'
          onChange={handleChangeChecked}
          checked={check}
        />
      </Form>

      {error && <ErrorDisplay error={error} resetError={resetError} />}
      <footer className={styles.footer}>@JosSid2022</footer>
    </div>
  );
};

export default LoginPage;
