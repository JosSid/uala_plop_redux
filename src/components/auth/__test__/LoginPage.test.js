import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import LoginPage from '../LoginPage.js';
import { defaultState} from '../../../store/reducers.js'

describe('LoginPage', () => {
  test('Snapshot', () => {
        const store = {
        getState: () => defaultState,
        dispatch: () => {},
        subscribe: () => {},
    }
    const {container} = render(
      <Provider store= {store}>
        <LoginPage />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
