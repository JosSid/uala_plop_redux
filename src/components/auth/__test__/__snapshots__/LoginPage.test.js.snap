// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage Snapshot 1`] = `
<div>
  <div
    class="loginPage"
  >
    <h1
      class="loginPage__title"
    >
      Welcome to
       
      <br />
      undefined
    </h1>
    <h4
      class="loginPage__title"
    >
      Please,login for search or create your advertisment
    </h4>
    <form
      class="loginPage__form"
      value="[object Object]"
    >
      <div
        class="formField loginForm__field"
      >
        <label
          class="formField__label"
        >
          <span>
            eMail
          </span>
          <input
            class="formField__input"
            name="email"
            type="text"
            value=""
          />
        </label>
      </div>
      <div
        class="formField loginForm__field"
      >
        <label
          class="formField__label"
        >
          <span>
            password
          </span>
          <input
            class="formField__input"
            name="password"
            type="password"
            value=""
          />
        </label>
      </div>
      <button
        class="sc-bcXHqe fTfPtG loginForm__submit"
        disabled=""
        type="submit"
      >
        Log in
      </button>
      <div>
        <label
          for="checklog"
          style="color: whitesmoke;"
        >
          Check for recording Login
        </label>
        <input
          name="checklog"
          type="checkbox"
        />
      </div>
    </form>
    <footer
      class="footer"
    >
      @JosSid2022
    </footer>
  </div>
</div>
`;
